CRM/Contribute/Form/ContributionPage/Amount.php

```diff
@@ -87,22 +87,28 @@ public function buildQuickForm() {
 
     $this->addElement('checkbox', 'is_monetary', ts('Execute real-time monetary transactions'));
 
-    $paymentProcessor = CRM_Core_PseudoConstant::paymentProcessor();
-    $recurringPaymentProcessor = array();
-
-    if (!empty($paymentProcessor)) {
-      $paymentProcessorIds = implode(',', array_keys($paymentProcessor));
-      $query = "
-SELECT id
-  FROM civicrm_payment_processor
- WHERE id IN ({$paymentProcessorIds})
-   AND is_recur = 1";
-      $dao = CRM_Core_DAO::executeQuery($query);
-      while ($dao->fetch()) {
-        $recurringPaymentProcessor[] = $dao->id;
+    $paymentProcessors = CRM_Financial_BAO_PaymentProcessor::getAllPaymentProcessors('live');
+    $recurringPaymentProcessor = $futurePaymentProcessor = $paymentProcessor = array();
+
+    if (!empty($paymentProcessors)) {
+      foreach ($paymentProcessors as $id => $processor) {
+        if ($processor['name'] != 'pay_later') {
```

Comment:
Actually, I removed the assignment to $this-> _values rather than $this-> _params. As I mentioned in the comment below, as long as the Recur records are being created using the processRecurringContribution  function, contrubuteRecurID will be set in $this->_params.